jepsen: the Register test is improperly configured

From https://teamcity.cockroachdb.com/viewLog.html?tab=buildLog&logTab=tree&filter=debug&expand=all&buildId=243251#_focus=3362

(Detected using new scripts from #15717 )

```
ERROR [2017-05-06 01:46:36,330] main - jepsen.cli Oh jeez, I'm sorry, Jepsen broke. Here's why:
java.util.concurrent.ExecutionException: java.lang.AssertionError: Assert failed: This jepsen.independent/concurrent-generator has 5 threads to work with, but can only use 0 of those threads to run 0 concurrent keys with 10 threads apiece. Consider raising or lowering the test's :concurrency to a multiple of 10.
(= thread-count (* group-size group-count))
```

The error occurs for any combination of nemeses.

--- a/build/teamcity-jepsen-run-one.sh
+++ b/build/teamcity-jepsen-run-one.sh
@@ -16,7 +16,8 @@ mkdir -p "${artifacts_dir}"
 # What is the controller again?
 controller=$(terraform output controller-ip)
 
-# Prepare the command to run the test.
+# Prepare the command to run the test. Note that --concurrency must be
+# a multiple of 10 and some tests require a minimum of 20.
 testcmd="cd jepsen/cockroachdb && set -eo pipefail && \
  stdbuf -oL -eL \
  ~/lein run test \
@@ -26,6 +27,7 @@ testcmd="cd jepsen/cockroachdb && set -eo pipefail && \
    --nodes-file ~/nodes \
    --os ubuntu \
    --time-limit 180 \
+   --concurrency 30 \
    --recovery-time 25 \
    --test-count 1 \
    --test ${test} ${nemesis} \

